### DISCUSSION TITLE
CommandLine Process ignoreCommand accepts only int for second value?

### DISCUSSION THREAD

**Discussion by Practical-UVM-Step-By-Step** [MikePopoloski/slang#869] [General]
Mike:
In your code in CommandLine.CPP we have:
```
std::string CommandLine::addIgnoreCommand(std::string_view value) {
....
    std::string error;
    auto result = parseInt<int>("", numArgs, error);
    if (result) {
        cmdIgnore[std::string(value)] = *result;
        return {};
    }
    return error;
}
```
Was trying to add some vendor specific switches to this list and noticed that it only accepts integers. 
Many vendors have the +warn=<some_string> or -warn=<some_string> or +/- error=<some_string>

Can we get an API that allows us to add a few vendor switches which have second argument as a string that updates your data structures?

I can possibly do this by extending the CommandLineClass and adding a method, 
but have a feeling a couple of other methods need to change in this class also, but thought I would mention this to you first and ask an opinion..

**Comment by MikePopoloski** [comment]
The integer being parsed here is the number of arguments to ignore following the command, not the name of the command itself, so you'd do something like `--cmd-ignore warn,1`

See docs here: https://sv-lang.com/command-line-ref.html#vendor-cmds

**Comment by Practical-UVM-Step-By-Step** [comment]
That was my point.. IT was kind of hard to do each one. I wound up changing the Driver.cpp in a fork.

Inside the Driver.cpp I did this so that I could keep the command line simple. 

cmdLine.addIgnoreCommand("-sverilog,0");
cmdLine.addIgnoreCommand("-warn=EDIBC,0");
cmdLine.addIgnoreCommand("-warn=TBIMN,0");
cmdLine.addIgnoreCommand("+error=COVCGI,0");

...
...
and so on.  I couldnt do -warn=*,0 because that section didnt seem to support regular expressions. (at least it didnt work anyways)
I was playing whack-a-mole for a while to get it to recognize a whole load of these because differnt command files I had were incorporating different ones and I was aggregating them all. 

I got this somewhat working in the fork and that's why I thought I would ask if there was a plan to leave this area alone.
I know i'll hit others as I test this parser on other designs...

**Reply by MikePopoloski** [reply]
The idea is not that you add an ignore for every possible parameter value, just an ignore for the command name itself. So an ignore for -warn, one for +error, etc. I think the cmd ignore and rename system doesn't support the =value format though, so right now it doesn't work. Should be simple to change that though. Someone else contributed this vendor command ignore feature; I'm not super thrilled with trying to be compatible with the exact same command line you use for literally all other possible tools, but in this case the fix should be easy.

**Reply by MikePopoloski** [reply]
Should be fixed in a50c4a37c1e0d1a5b6c0efecc2bcfb0127d14831
